scylla-cdc/src/consumer.rs

```diff
@@ -0,0 +1,485 @@
+        let mut data: Vec<Option<CqlValue>> = Vec::with_capacity(data_count);
+
+        for (i, column) in row.columns.into_iter().enumerate() {
+            if i == schema.stream_id {
+                stream_id = column.unwrap().into_blob().unwrap();
+            } else if i == schema.time {
+                time = column.unwrap().as_uuid().unwrap();
+            } else if i == schema.batch_seq_no {
+                batch_seq_no = column.unwrap().as_int().unwrap();
+            } else if i == schema.end_of_batch {
+                end_of_batch = column.unwrap().as_boolean().unwrap()
+            } else if i == schema.operation {
+                operation = OperationType::try_from(column.unwrap().as_tinyint().unwrap()).unwrap();
+            } else if i == schema.ttl {
+                ttl = column.map(|ttl| ttl.as_bigint().unwrap());
+            } else {
+                data.push(column);
+            }
+        }
+
+        CDCRow {
+            stream_id,
+            time,
+            batch_seq_no,
+            end_of_batch,
+            operation,
+            ttl,
+            data,
+            schema,
+        }
+    }
+
+    /// Allows to get a value from the column that corresponds to the logged table.
+    /// Returns None if the value is null.
+    /// Panics if the column does not exist in this table.
+    /// To check if such column exists, use column_exists() method.
+    pub fn get_value(&self, name: &str) -> &Option<CqlValue> {
+        self.schema
+            .mapping
+            .get(name)
+            .map(|id| &self.data[*id])
+            .unwrap()
+    }
+
+    /// Allows to get info if a value was deleted in this operation.
+    /// Panics if the column does not exist in this table
+    /// or the column is a part of primary key (because these values can't be deleted).
+    /// To check if such column exists, use column_deletable() method.
+    pub fn is_value_deleted(&self, name: &str) -> bool {
+        self.schema
+            .deleted_mapping
+            .get(name)
+            .map(|id| self.data[*id].is_some())
+            .unwrap()
+    }
+
+    /// Allows to get deleted elements from a collection.
+    /// Returns empty slice if the value is null.
+    /// Panics if the column does not exist in this table or is not a collection.
+    /// To check if such column exists, use collection_exists() method.
+    pub fn get_deleted_elements(&self, name: &str) -> &[CqlValue] {
+        let val = self
+            .schema
+            .deleted_el_mapping
+            .get(name)
+            .map(|id| self.data[*id].as_ref().map(|val| val.as_set().unwrap()))
+            .unwrap();
+        match val {
+            Some(vec) => vec,
+            None => &[],
+        }
+    }
+
+    pub fn column_exists(&self, name: &str) -> bool {
+        self.schema
+            .mapping
+            .get(name)
+            .map(|id| &self.data[*id])
+            .is_some()
```

Comment:
The `map` here is completely unnecessary. It doesn't change the result of the latter `is_some()` in any way, and the `&self.data[*id]` is not used at all.

Instead, you can probably do the following:

```rust
self.schema
    .mapping
    .contains_key(name)
```

The `column_deletable` and `collection_exists` methods have the same problem.